semtypes/src/main/java/io/ballerina/semtype/UniformTypeCode.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.semtype;
+
+/**
+ * Represent bit field that indicate which uniform type a semType belongs to.
+ * Regular types are divided longo mutable part and immutable part and these parts are called an uniform type.
+ * 5th bit indicate mutability; 0 immutable, 1 mutable
+ *
+ * @since 2.0.0
+ */
+public class UniformTypeCode {
+    // Inherently immutable
+    public static final long UT_NIL = 0x00;
```

Comment:
These should all be int. UniformTypeBitSet is a Unsigned32, but int is fine too.